assets/js/script.js

```diff
@@ -0,0 +1,210 @@
+// Assignment Code
+var generateBtn = document.querySelector("#generate");
+
+// Write password to the #password input
+function writePassword() {
+  var password = generatePassword();
+  var passwordText = document.querySelector("#password");
+
+  passwordText.value = password;
+
+}
+
+// Add event listener to generate button
+generateBtn.addEventListener("click", writePassword);
+
+//my code starts here
+
+//these are my criteria arrays
+const lowerCase = [
+  "a",
+  "b",
+  "c",
+  "d",
+  "e",
+  "f",
+  "g",
+  "h",
+  "i",
+  "j",
+  "k",
+  "l",
+  "m",
+  "n",
+  "o",
+  "p",
+  "q",
+  "r",
+  "s",
+  "t",
+  "u",
+  "v",
+  "w",
+  "x",
+  "y",
+  "z",
+];
+
+const upperCase = [
+  "A",
+  "B",
+  "C",
+  "D",
+  "E",
+  "F",
+  "G",
+  "H",
+  "I",
+  "J",
+  "K",
+  "L",
+  "M",
+  "N",
+  "O",
+  "P",
+  "Q",
+  "R",
+  "S",
+  "T",
+  "U",
+  "V",
+  "W",
+  "X",
+  "Y",
+  "Z",
+];
+
+const numeric = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+const specialChars = [
+  " ",
+  "!",
+  '"',
+  "#",
+  "$",
+  "%",
+  "&",
+  "'",
+  "(",
+  ")",
+  "*",
+  "+",
+  ",",
+  "-",
+  ".",
+  "/",
+  ":",
+  ";",
+  "<",
+  "=",
+  ">",
+  "?",
+  "@",
+  "[",
+  "\\",
+  "]",
+  "^",
+  "_",
+  "`",
+  "{",
+  "|",
+  "}",
+  "~",
+];
+
+// get criteria function
+const getCriteria = function () {
+  const passwordLength = Number(prompt("Choose a password length", ""));
+
+  //I need to check here that the input is in the range
+  if (passwordLength >= 8 && passwordLength <= 128) {
+    const useLowercase = confirm("Would you like to include lower case?");
+
+    const useUppercase = confirm("Would you like to use upper case?");
+
+    const useNumeric = confirm("Would you like to include numbers?");
+
+    const useSpecialCharacters = confirm(
+      "Would you like to include special characters?"
+    );
+
+    // at this point I need to see that not all have been answered false
+    if (
+      !useLowercase &&
+      !useUppercase &&
+      !useNumeric &&
+      !useSpecialCharacters
+    ) {
+      alert("You need to choose something to be in your password!");
```

Comment:
Being super picky here, maybe define 'something', e.g. "You must choose one of the criteria in order to generate a password."

Reply:
sorted, thank you